src/basic/smack-util.c

```diff
@@ -45,6 +45,75 @@ bool mac_smack_use(void) {
 #endif
 }
 
+static int mac_get_xattr(const char *path, const char *attr, const char **label) {
+        char *l = NULL;
+        ssize_t s;
+
+        assert(path);
+        assert(attr);
+        assert(label);
+
+        l = new0(char, NAME_MAX);
+        if (!l)
+                return -ENOMEM;
+
+        s = getxattr(path, attr, l, NAME_MAX);
```

Comment:
Wouldn't it be easier to use a stack allocated buffer here and then do a strdup() afterwards? That would make the code shorter and more readable.